### PR TITLE
feat(contracts-communication): remove batching

### PR DESCRIPTION
**Pull request by ChiTimesChi**
**Description**
A clear and concise description of the features you're adding in this pull request.

**Additional context**
Add any other context about the problem you're solving.

**Metadata**
- Fixes #[Link to Issue]

## Summary by CodeRabbit

- **Refactor**
  - Transitioned terminology and functionality from "batch" to "entry" across multiple contracts and interfaces, enhancing clarity and consistency.
  - Simplified function parameters and signatures to streamline interchain communication processes.

- **New Features**
  - Introduced new functions for encoding and decoding entry data, facilitating improved data handling and verification processes.

- **Bug Fixes**
  - Adjusted event emissions and function calls to align with the updated data structure, ensuring correct system behavior and interaction.

- **Documentation**
  - Updated comments and documentation to reflect changes from batch to entry verification and processing, aiding in better understanding and maintenance.

### CODE REVIEW

File: packages/contracts-communication/contracts/events/InterchainDBEvents.sol [hunk 1]
```diff
@@ -2,36 +2,26 @@
 pragma solidity ^0.8.0;
 
 abstract contract InterchainDBEvents {
-    /// @notice Emitted when a local entry is written to the database.
-    /// @param dbNonce      The nonce of the batch containing the entry.
-    /// @param entryIndex   The index of the entry within the batch.
+    /// @notice Emitted when a local entry (a specific digest by the specific writer) is written to the database.
```
Comment (aureliusbtc) [inline packages/contracts-communication/contracts/events/InterchainDBEvents.sol:5]: > local entry (a specific digest by the specific writer) 

should be simplified

File: packages/contracts-communication/contracts/events/InterchainDBEvents.sol [hunk 2]
```diff
@@ -2,36 +2,26 @@
...
+    /// @param dbNonce      The nonce of written entry.
     /// @param srcWriter    The address of the writer.
```
Comment (ChiTimesChi) [inline packages/contracts-communication/contracts/events/InterchainDBEvents.sol:7]: @aureliusbtc does this sound better?
```suggestion
    /// @notice Emitted when a local entry is written to the database.
    /// @param dbNonce      The nonce of the written entry.
    /// @param srcWriter    The address of the entry writer.
```

File: packages/contracts-communication/contracts/events/InterchainDBEvents.sol (hunk 1 above)
Comment (ChiTimesChi) [reply packages/contracts-communication/contracts/events/InterchainDBEvents.sol:5]: https://github.com/synapsecns/sanguine/pull/2599/files#r1595405516

Comment (ChiTimesChi) [reply packages/contracts-communication/contracts/events/InterchainDBEvents.sol]: Fixed in 2cb17d7b9544a80d8cc9700df20f8b2264671f17

**Pull request merged**